### ISSUE TITLE
Possibly incorrect error with explicit conversion of stackalloc to pointer

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#57040]
```c#
unsafe
{
    var v = (int*)stackalloc int[8];
}
```

Reports: 

`Conversion of a stackalloc expression of type 'int' to type 'int*' is not possible.`

Unclear if this is a bug.  Without the explicit cast, the stackalloc is treated as an `int*` and the assignment is fine.  It's unclear why explicitly casting to the type hte compiler implicitly converts this to is disallowed.

Seeking compiler clarification if this is intended (And if so, why).  Or if it is a bug.

**Comment by RikkiGibson** [comment]
notably `int* v = stackalloc int[8];` is accepted.

**Comment by jaredpar** [comment]
Lacking any special reasons I'm unaware of this does appear to be a bug. When evaluating this we should consider cases like `(Span<int>)stackalloc int[8]`
